core/store/src/trie/mod.rs

```diff
@@ -577,7 +573,6 @@ impl Trie {
         if *hash == Trie::empty_root() {
             Ok(memory.store(TrieNodeWithSize::empty()))
         } else {
-            self.counter.increment();
```

Comment:
Do we have any protocol changes here? In particular, I see there's `self.storage.retrieve_raw_bytes(hash)?;` in `delete_value`, and that now would increment the counter.

Reply:
Great question!

From what I see, all logic leading to incrementing counter here happens in `Trie::update` which itself is called after all tx/receipt execution. I feel that some counter increments makes us think that we charge for them somewhere, although we don't do it in reality. So it's better to remove them. But I also will be happy if someone double-check this.

Reply:
Is there a way to test this?

Reply:
@Longarithm I still don't understand why the change in `delete_value` won't cause any change in gas counting. Could you elaborate more on that?

One way to test this manually, is maybe run state viewer apply chain on some mainnet blocks to verify that gas counting doesn't change.

Reply:
I think `delete_value` will change here, because previously it only calls `self.storage.retrieve_raw_bytes(hash)?` without incrementing the counter. After your change, since you moved the counter incrementing logic inside `storage.retrieve_raw_bytes`, `delete_value` will increase the counter. 

I'm not super familiar with the gas counting logic, so I'm not sure why it is called in `Trie::update` won't affect gas counting.

Reply:
Yeah, you are right.

> I'm not sure why it is called in Trie::update won't affect gas counting.

I've looked over all `Trie::update` calls. It is called for four purposes:
* tests
* splitting states (`add_values_to_split_states_impl`)
* applying genesis (`finalize_genesis`)
* in `TrieUpdate::finalize`

The latter function is called for similar purposes - I think it doesn't make much sense to list all them. Also it is called in the end of `Runtime::apply` to apply changes to trie, when all tx/receipts are already executed and burnt/used gas is computed.

Moreover, trie counter is used for gas counting only during function call execution, which is triggered only by `process_receipt`, this is another reason why `Trie::update` change doesn't affect it.

Sorry I didn't explain it before properly. On my side, it was a fast look over CLion usages, so I thought it didn't worth attention.